src/Orleans.CodeGenerator/SerializerGenerator.cs

```diff
@@ -341,7 +342,10 @@ private static BaseCodecFieldDescription GetBaseTypeField(ISerializableTypeDescr
             if (type.SupportsPrimaryConstructorParameters)
             {
                 AddSerializationMembers(type, serializerFields, members.Where(m => m.IsPrimaryConstructorParameter), libraryTypes, writerParam, instanceParam, previousFieldIdVar, body);
-                body.Add(ExpressionStatement(InvocationExpression(writerParam.Member("WriteEndBase"), ArgumentList())));
+                if (body.LastOrDefault() != writeEndBaseInvocationExpression)
```

Comment:
This seems a little fragile - if some unrelated code is inserted in between then this check would break